lib/lightning/accounts.ex

```diff
@@ -492,44 +494,55 @@ defmodule Lightning.Accounts do
 
   ## Examples
 
-      iex> deliver_update_email_instructions(user, current_email, &Routes.user_update_email_url(conn, :edit, &1))
+      iex> deliver_update_email_instructions(user, new_email, &Routes.user_update_email_url(conn, :edit, &1))
       {:ok, %{to: ..., body: ...}}
 
   """
   def deliver_update_email_instructions(
         %User{} = user,
-        current_email,
+        new_email,
         update_email_url_fun
       )
       when is_function(update_email_url_fun, 1) do
     {encoded_token, user_token} =
-      UserToken.build_email_token(user, "change:#{user.email}", current_email)
+      UserToken.build_email_token(user, "change:#{user.email}", new_email)
 
     Repo.insert!(user_token)
 
     UserNotifier.deliver_update_email_warning(
       user,
-      current_email
+      new_email
     )
 
     UserNotifier.deliver_update_email_instructions(
-      user,
+      %User{
+        email: new_email,
+        first_name: user.first_name,
+        last_name: user.last_name
+      },
       update_email_url_fun.(encoded_token)
     )
   end
 
-  def validate_change_user_email(user, params \\ %{}) do
+  def validate_change_user_email(user, params \\ %{}, opts \\ []) do
+    validate_password? = Keyword.get(opts, :validate_password, false)
```

Comment:
Why do we conditionally validate the password?

Reply:
@stuartc the only reason why we are validating conditionally is to avoid validating the password after every keystroke. So I introduced the `validate_password` flag to be able to use the same function during form validation function and during form submission.